### PR TITLE
Add new whitespace-delimited example file and update examples for `read_table()` and `melt_*()`

### LINKED ISSUE #1333: Example dataset for `read_table()`
`massey-rating.txt` and `epa78.txt` aren't great datasets to use in the examples of `read_table()` in its current form (there are problems with variable number of fields). The use made sense in the past when `read_table()` worked more like what is now `read_fwf()`, e.g. it would calculate a fixed width and use it for all the fields.

A short term measure is to demo `read_table()` with `fwf-sample.txt`, but longer term it would be good to add a new example file that is not formatted with fixed-width fields, but that is whitespace delimited.

### PR DESCRIPTION
**Pull request by sbearrows**
Closes #1333

### CODE REVIEW

File: inst/extdata/whitespace-sample.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+firstName      lastName       state        phoneNumber
+John   Smith                WA        418-Y11-4111
+Mary               Hartford       CA         319-Z19-4341
+Evan Nolan          IL         219-532-c301
```
Comment (jennybc) [inline inst/extdata/whitespace-sample.txt:4]: This is on the right track.

I propose using column names that don't raise the question of camelCase vs. snake_case: use something more like `first`, `last`, `state`, `phone`.

Then I would also just use 1 space everywhere, maybe? Right now, the whitespace feels sort of random. It's not constant (e.g. 1 space) but it's also not aligned in any obvious way.

**Review by jennybc — COMMENTED** [review]
I made a few inline comments.

I recommend updating any `melt_*()` function to wherever this ends up. Yeah those functions are slated to live elsewhere, but it's also not too hard to make sure the example code is sensible in the meantime.

I prefer `writeLines(read_lines(...))` over `cat(read_lines(...))`. This is already used in some places.

**Comment by jennybc** [comment]
Nudge me to help make sense of the snapshot test failure. We'll probably want to resolve that first, in `main`, then rebase this branch, to keep separate matters separate.

File: inst/extdata/whitespace-sample.txt (hunk 1 above)
Comment (sbearrows) [reply inst/extdata/whitespace-sample.txt:4]: I created the file this way to "show off" what `read_table()` could do. I liked that when I read it in with `read_fwf()` there were empty cells but with `read_table()` it read it in cleanly.

I can make a file that isn't as scattered but that still does the above if you think that'd be better.

File: R/melt_table.R [hunk 2]
```diff
@@ -25,16 +25,9 @@
...
-#' melt_table(massey)
-#'
-#' # Sample of 1978 fuel economy data from
-#' # http://www.fueleconomy.gov/feg/epadata/78data.zip
-#' epa <- readr_example("epa78.txt")
-#' cat(read_file(epa))
-#' melt_table(epa)
+#' ws <- readr_example("whitespace-sample.txt")
+#' writeLines(read_file(ws))
+#' melt_table(ws)
```
Comment (jennybc) [inline R/melt_table.R:30]:
```suggestion
#' fwf <- readr_example("fwf-sample.txt")
#' writeLines(read_lines(fwf))
#' melt_table(fwf)
#' 
#' ws <- readr_example("whitespace-sample.txt")
#' writeLines(read_lines(ws))
#' melt_table2(ws)
```

Comment (jennybc) [reply R/melt_table.R:30]: Remember the overall goal is to demonstrate readr functions with an example dataset that shows the most suitable input for the function. The discovery of #1333 is that these examples were originally written when readr behaved differently. Then the readr functions changed, but the examples didn't, which left behind some rather nonsensical example code.

The `melt*()` functions are designed to return output where "each row represents a single token".

I think `melt_table()` is best paired with `fwf-sample.txt`. `melt_table2()` makes much more sense for `whitespace-sample.txt`. 

Also, from playing around with this, I think `read_lines()` is generally what we want (not `read_file()`).

``` r
devtools::load_all()
#> ℹ Loading readr

fwf <- readr_example("fwf-sample.txt")
writeLines(read_lines(fwf))
#> John Smith          WA        418-Y11-4111
#> Mary Hartford       CA        319-Z19-4341
#> Evan Nolan          IL        219-532-c301
melt_table(fwf)
#> Warning: `melt_table()` was deprecated in readr 2.0.0.
#> Please use `meltr::melt_table()` instead
#> # A tibble: 12 × 4
#>      row   col data_type value       
#>    <dbl> <dbl> <chr>     <chr>       
#>  1     1     1 character John        
#>  2     1     2 character Smith       
#>  3     1     3 character WA          
#>  4     1     4 character 418-Y11-4111
#>  5     2     1 character Mary        
#>  6     2     2 character Hartford    
#>  7     2     3 character CA          
#>  8     2     4 character 319-Z19-4341
#>  9     3     1 character Evan        
#> 10     3     2 character Nolan       
#> 11     3     3 character IL          
#> 12     3     4 character 219-532-c301

ws <- readr_example("whitespace-sample.txt")
writeLines(read_lines(ws))
#> first last state phone
#> John Smith WA 418-Y11-4111
#> Mary Hartford CA 319-Z19-4341
#> Evan Nolan IL 219-532-c301

# this doesn't make much sense as an example
# it does not produce one token per row
melt_table(ws)
#> Warning: `melt_table()` was deprecated in readr 2.0.0.
#> Please use `meltr::melt_table()` instead
#> # A tibble: 4 × 4
#>     row   col data_type value                        
#>   <dbl> <dbl> <chr>     <chr>                        
#> 1     1     1 character first last state phone       
#> 2     2     1 character John Smith WA 418-Y11-4111   
#> 3     3     1 character Mary Hartford CA 319-Z19-4341
#> 4     4     1 character Evan Nolan IL 219-532-c301

# this does
melt_table2(ws)
#> Warning: `melt_table2()` was deprecated in readr 2.0.0.
#> Please use `meltr::melt_table2()` instead
#> # A tibble: 16 × 4
#>      row   col data_type value       
#>    <dbl> <dbl> <chr>     <chr>       
#>  1     1     1 character first       
#>  2     1     2 character last        
#>  3     1     3 character state       
#>  4     1     4 character phone       
#>  5     2     1 character John        
#>  6     2     2 character Smith       
#>  7     2     3 character WA          
#>  8     2     4 character 418-Y11-4111
#>  9     3     1 character Mary        
#> 10     3     2 character Hartford    
#> 11     3     3 character CA          
#> 12     3     4 character 319-Z19-4341
#> 13     4     1 character Evan        
#> 14     4     2 character Nolan       
#> 15     4     3 character IL          
#> 16     4     4 character 219-532-c301
```

<sup>Created on 2022-01-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

File: R/read_table.R [hunk 3]
```diff
@@ -17,16 +17,9 @@
...
-#' read_table(massey)
-#'
-#' # Sample of 1978 fuel economy data from
-#' # http://www.fueleconomy.gov/feg/epadata/78data.zip
-#' epa <- readr_example("epa78.txt")
-#' cat(read_file(epa))
-#' read_table(epa, col_names = FALSE)
+#' ws <- readr_example("whitespace-sample.txt")
+#' writeLines(read_file(ws))
+#' read_table(ws, col_names = FALSE)
```
Comment (jennybc) [inline R/read_table.R:22]:
```suggestion
#' writeLines(read_lines(ws))
#' read_table(ws)
```

I think we absolutely want the `col_names` in this case. Also making the switch to `read_lines()`, as explained above.

``` r
devtools::load_all()
#> ℹ Loading readr

ws <- readr_example("whitespace-sample.txt")
writeLines(read_lines(ws))
#> first last state phone
#> John Smith WA 418-Y11-4111
#> Mary Hartford CA 319-Z19-4341
#> Evan Nolan IL 219-532-c301

# this produces incorrect column names and incorrect data
read_table(ws, col_names = FALSE)
#> 
#> ── Column specification ────────────────────────────────────────────────────────
#> cols(
#>   X1 = col_character(),
#>   X2 = col_character(),
#>   X3 = col_character(),
#>   X4 = col_character()
#> )
#> # A tibble: 4 × 4
#>   X1    X2       X3    X4          
#>   <chr> <chr>    <chr> <chr>       
#> 1 first last     state phone       
#> 2 John  Smith    WA    418-Y11-4111
#> 3 Mary  Hartford CA    319-Z19-4341
#> 4 Evan  Nolan    IL    219-532-c301

# this produces correct column names and correct data
read_table(ws)
#> 
#> ── Column specification ────────────────────────────────────────────────────────
#> cols(
#>   first = col_character(),
#>   last = col_character(),
#>   state = col_character(),
#>   phone = col_character()
#> )
#> # A tibble: 3 × 4
#>   first last     state phone       
#>   <chr> <chr>    <chr> <chr>       
#> 1 John  Smith    WA    418-Y11-4111
#> 2 Mary  Hartford CA    319-Z19-4341
#> 3 Evan  Nolan    IL    219-532-c301
```

<sup>Created on 2022-01-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

**Pull request merged**